core/src/main/java/org/mule/config/DefaultMuleConfiguration.java

```diff
@@ -276,119 +272,6 @@ else if (isStandalone())
-        if (p != null)
-        {
-            encoding = p;
-        }
-        else
-        {
-            System.setProperty(MuleProperties.MULE_ENCODING_SYSTEM_PROPERTY, encoding);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "endpoints.synchronous");
-        if (p != null)
-        {
-            synchronous = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "systemModelType");
-        if (p != null)
-        {
-            systemModelType = p;
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "timeout.synchronous");
-        if (p != null)
-        {
-            responseTimeout = NumberUtils.toInt(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "timeout.transaction");
-        if (p != null)
-        {
-            defaultTransactionTimeout = NumberUtils.toInt(p);
-        }
-
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "workingDirectory");
-        if (p != null)
-        {
-            workingDirectory = p;
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "clientMode");
-        if (p != null)
-        {
-            clientMode = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "serverId");
-        if (p != null)
-        {
-            id = p;
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "domainId");
-        if (p != null)
-        {
-            domainId = p;
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "message.cacheBytes");
-        if (p != null)
-        {
-            cacheMessageAsBytes = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "message.cacheOriginal");
-        if (p != null)
-        {
-            cacheMessageOriginalPayload = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "streaming.enable");
-        if (p != null)
-        {
-            enableStreaming = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "transform.autoWrap");
-        if (p != null)
-        {
-            autoWrapMessageAwareTransform = BooleanUtils.toBoolean(p);
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "stacktrace.full");
-        if (p != null)
-        {
-            fullStackTraces = false;
-        }
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "stacktrace.filter");
-        if (p != null)
-        {
-            stackTraceFilter = p.split(",");
-        }
-
-        p = System.getProperty(MuleProperties.SYSTEM_PROPERTY_PREFIX + "verbose.exceptions");
```

Comment:
Seems to me that the clean up should not delete applySystemProperties method, as this property is not configurable from the XML, so it does not going to be overwritten when the XML is parsed. Check every property configured in this method and delete only the ones that are configurable on the XML